src/ui/src/wds/WdsTextInput.vue

```diff
@@ -19,8 +41,28 @@ input {
 	background: transparent;
 }
 
-input:focus {
+.WdsTextInput:focus,
+.WdsTextInput--leftIcon:focus-within {
 	border: 1px solid var(--softenedAccentColor);
 	box-shadow: 0px 0px 0px 3px rgba(81, 31, 255, 0.05);
 }
+
+.WdsTextInput--leftIcon {
```

Comment:
Hey I guess nothing wrong with it but just wanted to know, why do you do 

`.WdsTextInput--leftIcon`

instead of

`.WdsTextInput.leftIcon`

It'd more compact since

`class="WdsTextInput WdsTextInput--leftIcon"`

becomes

`class="WdsTextInput leftIcon"`

Since the styles are scoped there's no risk of another `leftIcon` bothering you.

Reply:
It's just a habit of following [BEM convention](https://getbem.com/naming/). It helps to improve the organization of the CSS. So the `*--leftIcon` acts as a "modifier". A modifier means to change appearance, behavior or state of an element.

But I agree that here, it's not necessary since the style is scoped